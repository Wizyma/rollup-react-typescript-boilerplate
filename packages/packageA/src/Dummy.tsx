import { string } from 'prop-types';
import * as React from 'react';

type DummyProps = {
  title?: string
}

const DummyComponent: React.FC<DummyProps> = ({ title }: DummyProps): JSX.Element => (
  <h1>{title}</h1>
);

DummyComponent.propTypes = {
  title: string
}
DummyComponent.defaultProps = {
  title: 'Dummy Component!'
}

export default DummyComponent;
