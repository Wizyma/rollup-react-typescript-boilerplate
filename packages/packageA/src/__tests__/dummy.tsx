import render from 'react-test-renderer';
import * as React from 'react';

import Dummy from '../Dummy';

describe('Component should render', () => {
  it('should match  with custom prop', () => {
    const comp = render.create(<Dummy title="Other Title" />)
    expect(comp.toJSON()).toMatchSnapshot();
  })

  it('should match  with default props', () => {
    const comp = render.create(<Dummy />)
    expect(comp.toJSON()).toMatchSnapshot();
  })
})
