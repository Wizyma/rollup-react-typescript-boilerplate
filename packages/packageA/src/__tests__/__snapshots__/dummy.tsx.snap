// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render should match  with custom prop 1`] = `
<h1>
  Other Title
</h1>
`;

exports[`Component should render should match  with default props 1`] = `
<h1>
  Dummy Component!
</h1>
`;
